spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/UDTValue.scala

```diff
@@ -6,24 +6,22 @@ import scala.reflect.runtime.universe._
 import com.datastax.driver.core.{UDTValue => DriverUDTValue}
 import com.datastax.spark.connector.types.NullableTypeConverter
 
-final case class UDTValue(columnNames: IndexedSeq[String], columnValues: IndexedSeq[AnyRef])
+final case class UDTValue(metaData: CassandraRowMetadata, columnValues: IndexedSeq[AnyRef])
```

Comment:
@jtgrabowski , as I changed the case class fields, the constructor and the default apply and unapply methods changed.
Also the getter for columnNames get disappeared. This breaks compatibility. I've seen the usage example for this class in the docs: UDTValue.fromMap . Nothing prohibits this class from being instantiated directly.

Please let me know, if we should take care of compatibility here.
Maybe Option[CassandraRowMetadata] as 3rd parameter could help here.

Thanks